examples/permissioned-token/test/failing-paths.js

```diff
@@ -0,0 +1,178 @@
+const { types } = require('@algo-builder/runtime');
+const { assert } = require('chai');
+const {
+  optInToASA,
+  optInToPermissions,
+  issue,
+  whitelist,
+  fund,
+  killToken,
+  transfer,
+  optOut,
+  forceTransfer,
+  setupEnv
+} = require('./common');
+
+describe('Permissioned Token Tests -- Failing Paths', function () {
+  let runtime, master, alice, bob, elon;
+  let lsig, assetIndex, controllerAppID, permissionsAppId;
+
+  // Sync Accounts and ASA def
+  function syncAccounts () {
+    master = runtime.getAccount(master.address);
+    alice = runtime.getAccount(alice.address);
+    bob = runtime.getAccount(bob.address);
+    elon = runtime.getAccount(elon.address);
+  }
+
+  this.beforeEach(async function () {
+    // Create Accounts and Env
+    [
+      runtime,
+      master,
+      alice,
+      bob,
+      elon,
+      assetIndex,
+      controllerAppID,
+      lsig,
+      permissionsAppId
+    ] = setupEnv();
+  });
+
+  it('Token Issuance', () => {
+    // Cannot issue before opting-in
+    assert.throws(() => issue(runtime, alice, elon, 20, controllerAppID, assetIndex, lsig), 'RUNTIME_ERR1404');
+
+    // Opting-in
+    optInToASA(runtime, elon.address, assetIndex);
+    syncAccounts();
+
+    // Cannot issue after killing token
+    killToken(runtime, alice, controllerAppID);
+    assert.throws(() => issue(runtime, alice, elon, 20, controllerAppID, assetIndex, lsig), 'RUNTIME_ERR1009');
+  });
+
+  it('Kill Token', () => {
+    // Fails as wrong ASA manager is passed
+    assert.throws(() => killToken(runtime, bob, controllerAppID), 'RUNTIME_ERR1009');
+  });
+
+  it('WhiteListing', () => {
+    // Only asset manager can whitelist
+    const address = elon.address;
+    optInToPermissions(runtime, address, permissionsAppId);
+    syncAccounts();
+    assert.isDefined(elon.getAppFromLocal(permissionsAppId));
+
+    // Fails because Bob is not the manager
+    assert.throws(() => runtime.executeTx({
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: bob.account, // Bob is not the asset manager
+      appId: permissionsAppId,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:add_whitelist', `int:${controllerAppID}`],
+      accounts: [address],
+      foreignAssets: [assetIndex],
+      foreignApps: [controllerAppID]
+    }), 'RUNTIME_ERR1009');
+  });
+
+  it('Opt Out', () => {
+    // Opt-out wont work without opting in
+    assert.throws(() => optOut(runtime, alice, elon, assetIndex), 'RUNTIME_ERR1404');
```

Comment:
If we add optOut params here, then i think it will be more clear, as it is only a single transaction. What do you think?

Reply:
The helper function is being used at several instances, if we directly make the transaction here it will add redundant code.